universal-login-web3/src/ULWeb3Provider.ts

```diff
@@ -238,12 +234,16 @@ export class ULWeb3Provider implements Provider {
   }
 
   finalizeAndStop() {
+    return this.initHandler.finalize();
+  }
+
+  private _finalizeAndStop() {
+    this.walletService.stateProperty.set({kind: 'None'});
```

Comment:
Is this needed?

Also setting the state property directly kinda breaks encapsulation. We should make the setter private so only `WalletService` can mutate it's state.